### PR TITLE
Remove openssl pins

### PR DESCRIPTION
**Pull request by h-vetinari**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,9 +34,7 @@ requirements:
 
 test:
   requires:
-    # add explicit tests for compatibility for openssl 3 and 1.1.1
-    - openssl 3   # [not linux64]
-    - openssl 1   # [linux64]
+    - openssl
```
Comment (hmaarrfk) [inline recipe/meta.yaml]: You can probably remove the requires line alltogether at this point then.

I was mostly debugging things.

Comment (hmaarrfk) [reply recipe/meta.yaml]: I think this is the ultimate PR that fixed things for me https://github.com/conda-forge/libssh-feedstock/pull/15/files

Comment (h-vetinari) [reply recipe/meta.yaml]: I mean, I wanted to make sure that we're definitely passing with both OpenSSL versions. But happy to remove the pins altogether.

Comment (hmaarrfk) [reply recipe/meta.yaml]: the old pin was doing that though. what was wrong with it?

Comment (h-vetinari) [reply recipe/meta.yaml]: No, it wasn't. It was testing one platform with openssl 1, and the others with openssl 3. My previous iteration here made sure that all platforms were tested against both.

We should choose one approach that's consistent across platforms
- test everything against only OpenSSL 3
- test everything against Openssl 1.1.1 & 3

In any case, the hard pins in the recipe should go.

Comment (hmaarrfk) [reply recipe/meta.yaml]: I don't recall the specifics of this previous iteration but i do think we shohld keep the test matrix under control.

The attempt to test openssl different versins on different platforms was to ensure the test matrix doesn't increase in total size, but rather increase its reach.

It's somewhat a similar concept to random sampling vs regular sampling.

Comment (h-vetinari) [reply recipe/meta.yaml]: > The attempt to test openssl different versins on different platforms was to ensure the test matrix doesn't increase in total size, but rather increase its reach.

I get the intention, but it's IMO not worth having silent stumbling blocks like that (e.g. if we had switched to OpenSSL 3, this feedstock would not have been tested correctly anymore)

**Pull request merged**